lib/simplecov/result_merger.rb

```diff
@@ -62,7 +62,7 @@ def results
       def merged_result
         merged = {}
         results.each do |result|
-          merged = result.original_result.merge_resultset(merged)
+          merged = SimpleCov::MergeHelper.merge_resultsets(result.original_result, merged)
```

Comment:
I had actually been thinking about adding a method that takes an array of `SimpleCov::Result` and merges them. ResultMerger does this internally but doesn't expose it as an API. That could certainly build on the API you've proposed here.

Earlier you mentioned putting this code in ResultMerger, but now it sounds like you'd like it in a new `RawCoverage` module instead? Happy to do whatever seems best to you.